src/main/java/com/amazon/opendistroforelasticsearch/security/resolver/IndexResolverReplacer.java

```diff
@@ -359,26 +329,24 @@ public Resolved resolveRequest(final Object request) {
          */
         private static final Set<String> All_SET = Collections.singleton("*");
         private static final long serialVersionUID = 1L;
-        public final static Resolved _LOCAL_ALL = new Resolved(All_SET, All_SET, All_SET, All_SET, Collections.emptySet(), Collections.emptySet());
+        public final static Resolved _LOCAL_ALL = new Resolved(All_SET, All_SET, All_SET, Collections.emptySet(), Collections.emptySet());
         private final Set<String> aliases;
-        private final Set<String> indices;
         private final Set<String> allIndices;
         private final Set<String> types;
 
         private final Set<String> originalRequested;
         private final Set<String> remoteIndices;
 
-        private Resolved(final Set<String> aliases, final Set<String> indices, final Set<String> allIndices,
+        private Resolved(final Set<String> aliases, final Set<String> allIndices,
                          final Set<String> types, final Set<String> originalRequested, final Set<String> remoteIndices) {
             super();
             this.aliases = aliases;
-            this.indices = indices;
             this.allIndices = allIndices;
             this.types = types;
```

Comment:
Hardcode `type` to `ImmutableSet.of("*")`;

Reply:
Done, I still read/write from stream to avoid breaking protocol compatibility.

Reply:
As far as I can see `Resolved` is transient and is never serialized/de-serialized.